granulate_utils/linux/elf.py

```diff
@@ -12,6 +13,20 @@
 __all__ = ["ELFError"]
 
 
+def catch_filenotfound(func):
```

Comment:
Nice approach! (decorator).

A few suggestions:
1. I see that this will also fix the problem for `get_mapped_dso_elf_id` - good.
2. I suggest a more descriptive name - maybe `raise_nosuchprocess`? That's more important than the `catch filenotfound` part.
3. This is not robust enough. For example, it could raise `NoSuchProcess` if I try `get_elf_id("/proc/blablabla")` (where blablabla is not even a PID). It could also raise `NoSuchProcess` if I try `/proc/pid/root/path/doesnt/exist`, even if `pid` is running indeed. I suggest - in the decorator you check if the path starts with `/proc/`, and if so, is the 2nd component a number, and if so, is it running, and if not - raise.

Reply:
1. Done
2. Done
3. Done. Also added check if number from /proc/{number} is a number from PID number range. This avoids getting OverflowError from `psutil.pid_exists(pid)` for big numbers.

Reply:
The max PID check might be a bit of an overkill, but it's indeed as robust as we'll go. So okay :)

Reply:
No, it wasn't for an actual PID. I was just wondering what will happen if I try to get some really huge number instead of real PID (f.e. `get_elf_id("/proc/3276842132132323/exe")` -> resulted with OverflowError for `psutil.pid_exists(pid)`).